src/services/authService.ts

```diff
@@ -0,0 +1,55 @@
+import appConfig from "../config/appConfig";
+import { fetchUserData } from "./userService";
+
+//authURL
+export const constructAuthUrl = () => {
+  const authUrl = new URL(`${appConfig.wcaUrl}/oauth/authorize`);
+
+  // Configure OAuth parameters
+  authUrl.searchParams.append("client_id", appConfig.wcaClientId);
+  authUrl.searchParams.append("redirect_uri", "/api/auth/callback");
+  authUrl.searchParams.append("response_type", "code");
+
+  return authUrl.toString();
+};
+
+
+//loginURL
+export const contructLoginUrl = async (code: string) => {
+  const frontendUrl = new URL(appConfig.frontendUrl);
+
```

Comment:
We can add the pathname here itself because you are anyway having the same pathname everywhere else. Also you can rename the variable as `loginUrl`, and reuse the same object throughout the method instead of creating separate object everywhere.